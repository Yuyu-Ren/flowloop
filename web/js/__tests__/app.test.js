/* eslint-env jest */
import App from '../app'
import MockDate from 'mockdate'
import pretty from 'pretty'

jest.useFakeTimers()

let app

beforeEach(() => {
  MockDate.set('4/20/2010', 120)
  app = new App({ env: 'test' })
  app.start()
})

afterEach(() => {
  MockDate.reset()
  app.stop()
})

describe('App', () => {
  test('starts and dies flawlessly', () => {
    expect(1).toEqual(1)
  })

  test('drops the right HTML', () => {
    expect(pretty(document.body.innerHTML)).toMatchSnapshot()
  })

  test('Empty timer', () => {
    app.store.dispatch({ type: 'router:nav!', to: '/timer' })
    jest.runOnlyPendingTimers()
    jest.runOnlyPendingTimers()
    expect(pretty(document.body.innerHTML)).toMatchSnapshot()
  })

  test('Started work timer', () => {
    app.store.dispatch({ type: 'timer:start!', timerType: 'work' })
    jest.runOnlyPendingTimers()
    jest.runOnlyPendingTimers()
    expect(pretty(document.body.innerHTML)).toMatchSnapshot()
  })

  test('Started break timer', () => {
    app.store.dispatch({ type: 'timer:start!', timerType: 'break' })
    jest.runOnlyPendingTimers()
    jest.runOnlyPendingTimers()
    expect(pretty(document.body.innerHTML)).toMatchSnapshot()
  })

  test('Empty log', () => {
    app.store.dispatch({ type: 'router:nav!', to: '/log' })
    jest.runOnlyPendingTimers()
    jest.runOnlyPendingTimers()
    expect(pretty(document.body.innerHTML)).toMatchSnapshot()
  })
})
