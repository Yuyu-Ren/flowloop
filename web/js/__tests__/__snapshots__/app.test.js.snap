// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Empty log 1`] = `
"<div id=\\"root\\" class=\\"app-root\\"><span data-reactroot=\\"\\"><div class=\\"timer-layout -log _page-bottom _page-enter\\">
<!-- react-empty: 24 --><div class=\\"rawbody\\"><div class=\\"slim-container\\"><div class=\\"actions-list\\"><div class=\\"right\\"><button class=\\"icon-button -close\\"></button></div></div><div class=\\"blank-state -log\\"><h2>Timeline</h2><p>When you finish a work period, it will appear here.</p></div></div></div></div><div class=\\"timer-layout _page-fade _page-leave\\"><div class=\\"topleft\\"><a class=\\"settings-button\\" href=\\"#settings\\"> <span class=\\"icon\\"></span>  </a> 
</div>
<div class=\\"body\\">

  <!-- react-empty: 7 -->
  <div class=\\"timer-chooser\\">
    <button class=\\"button timer-button -work -large\\">Work</button>
    <div class=\\"logo app-logo\\"></div>
    <button class=\\"button timer-button -break -large\\">Break</button>
  </div>
  <div class=\\"toast-base\\">
    <div class=\\"item\\">
      <div class=\\"toast-item _toast-pop-in\\">
        <p>Flowloop is a productivity timer that helps you maintain a state of flow.</p>
        <div class=\\"actions\\">
          <button class=\\"action \\">Learn more</button>
          <button class=\\"action -mute\\">Dismiss</button>
        </div>
      </div>
    </div>
  </div>
</div>
<a class=\\"timeline-summary fixed\\" href=\\"#log\\"><span class=\\"-left text\\">Tuesday</span> <span class=\\"triangle\\"></span>  <span class=\\"-right text\\">Apr 20</span>  </a> 
</div>
</span> 
</div>"
`;

exports[`App Empty timer 1`] = `
"<div id=\\"root\\" class=\\"app-root\\"><span data-reactroot=\\"\\"><div class=\\"timer-layout _page-fade\\"><div class=\\"topleft\\"><a class=\\"settings-button\\" href=\\"#settings\\"> <span class=\\"icon\\"></span>  </a> 
</div>
<div class=\\"body\\">

  <!-- react-empty: 7 -->
  <div class=\\"timer-chooser\\">
    <button class=\\"button timer-button -work -large\\">Work</button>
    <div class=\\"logo app-logo\\"></div>
    <button class=\\"button timer-button -break -large\\">Break</button>
  </div>
  <div class=\\"toast-base\\">
    <div class=\\"item\\">
      <div class=\\"toast-item _toast-pop-in\\">
        <p>Flowloop is a productivity timer that helps you maintain a state of flow.</p>
        <div class=\\"actions\\">
          <button class=\\"action \\">Learn more</button>
          <button class=\\"action -mute\\">Dismiss</button>
        </div>
      </div>
    </div>
  </div>
</div>
<a class=\\"timeline-summary fixed\\" href=\\"#log\\"><span class=\\"-left text\\">Tuesday</span> <span class=\\"triangle\\"></span>  <span class=\\"-right text\\">Apr 20</span>  </a> 
</div>
</span> 
</div>"
`;

exports[`App Started break timer 1`] = `
"<div id=\\"root\\" class=\\"app-root\\"><span data-reactroot=\\"\\"><div class=\\"timer-layout _page-fade\\"><div class=\\"topleft\\"><a class=\\"settings-button\\" href=\\"#settings\\"> <span class=\\"icon\\"></span>  </a> 
</div>
<div class=\\"body\\">

  <!-- react-empty: 28 -->
  <div class=\\"timer-controls\\">
    <div class=\\"title\\">
      <h1 class=\\"\\">
        <span class=\\"break-selector\\">Break</span> 
      </h1>
      <p class=\\"subtitle\\">

        <!-- react-text: 34 -->0:00<!-- /react-text -->

        <!-- react-text: 35 --><!-- /react-text --><span> of </span> 

        <!-- react-text: 37 --><!-- /react-text -->

        <!-- react-text: 38 -->5:00<!-- /react-text -->
      </p>
    </div>
    <div class=\\"actions\\"><span class=\\"popup-set\\"><span class=\\"trigger\\"><button aria-label=\\"Continuous\\" class=\\"mode-selector-button hint--bottom -continuous\\">  <span class=\\"icon\\"></span> </button>
      </span> 
      </span> 
    </div>
  </div>
  <div class=\\"timer-spacer\\"></div>
  <div class=\\"timer-display\\">
    <div class=\\"pie-graph pie _fade-in-slow\\"><span class=\\"fill\\" style=\\"background: rgb(198, 202, 208);\\"></span> <span class=\\"piece -fill\\" style=\\"background: rgb(198, 202, 208);\\"></span> <span class=\\"center\\"></span> </div><span class=\\"label\\">5:00</span> </div>
  <div class=\\"timer-spacer\\"></div>
  <div class=\\"timer-actions\\">
    <button aria-label=\\"Stop\\" class=\\"timer-button button -stop\\">  <span class=\\"icon\\"></span> </button>
  </div>
</div>
<a class=\\"timeline-summary fixed\\" href=\\"#log\\"><span class=\\"-left text\\">Tuesday</span> <span class=\\"triangle\\"></span>  <span class=\\"-right text\\">Apr 20</span>  </a> 
</div>
</span> 
</div>"
`;

exports[`App Started work timer 1`] = `
"<div id=\\"root\\" class=\\"app-root\\"><span data-reactroot=\\"\\"><div class=\\"timer-layout _page-fade\\"><div class=\\"topleft\\"><a class=\\"settings-button\\" href=\\"#settings\\"> <span class=\\"icon\\"></span>  </a> 
</div>
<div class=\\"body\\">

  <!-- react-empty: 67 -->
  <div class=\\"timer-chooser\\">
    <button class=\\"button timer-button -work -large\\">Work</button>
    <div class=\\"logo app-logo\\"></div>
    <button class=\\"button timer-button -break -large\\">Break</button>
  </div>
  <div class=\\"toast-base\\">
    <div class=\\"item\\">
      <div class=\\"toast-item _toast-pop-in\\">
        <p>Flowloop is a productivity timer that helps you maintain a state of flow.</p>
        <div class=\\"actions\\">
          <button class=\\"action \\">Learn more</button>
          <button class=\\"action -mute\\">Dismiss</button>
        </div>
      </div>
    </div>
  </div>
</div>
<a class=\\"timeline-summary fixed\\" href=\\"#log\\"><span class=\\"-left text\\">Tuesday</span> <span class=\\"triangle\\"></span>  <span class=\\"-right text\\">Apr 20</span>  </a> 
</div>
</span> 
</div>"
`;

exports[`App drops the right HTML 1`] = `
"<div id=\\"root\\" class=\\"app-root\\"><span data-reactroot=\\"\\"><div class=\\"timer-layout _page-fade\\"><div class=\\"topleft\\"><a class=\\"settings-button\\" href=\\"#settings\\"> <span class=\\"icon\\"></span>  </a> 
</div>
<div class=\\"body\\">

  <!-- react-empty: 7 -->
  <div class=\\"timer-chooser\\">
    <button class=\\"button timer-button -work -large\\">Work</button>
    <div class=\\"logo app-logo\\"></div>
    <button class=\\"button timer-button -break -large\\">Break</button>
  </div>
  <div class=\\"toast-base\\">
    <div class=\\"item\\">
      <div class=\\"toast-item _toast-pop-in\\">
        <p>Flowloop is a productivity timer that helps you maintain a state of flow.</p>
        <div class=\\"actions\\">
          <button class=\\"action \\">Learn more</button>
          <button class=\\"action -mute\\">Dismiss</button>
        </div>
      </div>
    </div>
  </div>
</div>
<a class=\\"timeline-summary fixed\\" href=\\"#log\\"><span class=\\"-left text\\">Tuesday</span> <span class=\\"triangle\\"></span>  <span class=\\"-right text\\">Apr 20</span>  </a> 
</div>
</span> 
</div>"
`;
